src/app/autocomplete-input/autocomplete-input.component.ts

```diff
@@ -62,21 +80,22 @@ export class AutocompleteInputComponent {
         this.schemaKeysStoreService.forPath(state.path).filter(item => item.startsWith(state.query))
       );
     });
-    this.valueChange$
-      .debounceTime(500)
-      .subscribe(value => {
-        this.valueChange.emit(value);
-      });
   }
 
-  modelChange(value: string) {
-    this.value = value;
-    this.valueChange$.next(this.value);
+  ngOnInit() {
+    if (this.tagsEnabled) {
+      this.mousetrap = Mousetrap(this.el.nativeElement);
+      this.mousetrap.bind(this.tagShortcut, (event) => {
+        event.preventDefault();
+        this.selectedValues = this.selectedValues.add(this.value);
+        this.value = '';
+        this.valueChange.emit(this.selectedValues.toJS());
```

Comment:
It's a bit optional but I think you can use it as `Immutable.Set` also on parent component and just convert it `toJS` just before going out of the app (such as http request to server)

Reply:
It doesn't go out of the app. I am using it to filter each record given the selected paths.